app/src/components/MealDetail.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+const MealDetail = () => {
+  const { id } = useParams();
+  const [meal, setMeal] = useState(null);
+  const [isAvailable, setIsAvailable] = useState(false);
+  const [formData, setFormData] = useState({
+    name: "",
+    email: "",
+    phone: "",
+  });
+
+  useEffect(() => {
+    // Fetch meal details by ID
+    fetch(`http://localhost:3001/api/meals/${id}`)
+      .then((response) => response.json())
+      .then((data) => {
+        setMeal(data);
+        setIsAvailable(data.availableReservations > 0);
+      });
+  }, [id]);
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (!isAvailable) {
+      alert("No reservations available for this meal");
+      return;
+    }
+
+    // Submit reservation
+    fetch("http://localhost:3001/api/reservations", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        mealId: id,
+        ...formData,
+      }),
+    })
```

Comment:
Fetch can also throw an error => some errors might not be caught with current code